p2p/swarmimpl.go

```diff
@@ -280,14 +284,52 @@ func (s *swarmImpl) shutDownInternal() {
 		c.Close()
 	}
 
+	s.peerMapMutex.Lock()
 	for _, p := range s.peers {
 		p.DeleteAllConnections()
 	}
 
+	s.peerMapMutex.Unlock()
 	// shutdown network
 	s.network.Shutdown()
 }
 
+func (s *swarmImpl) BlockUntilBoot() {
```

Comment:
There is no way to alert in case the bootstrap failed, I think that using struct{} is wrong here